Update CODE_OF_CONDUCT.md (#19812)
coq/coq: CODE_OF_CONDUCT.md

  constructively. Remember that we are different. Different people
  have different perspectives on issues. Being unable to understand why someone
  holds a viewpoint doesn't mean that they're wrong.

> - **Avoid excessively long messages.**
>   Long messages take more time for others to read and respond to, and
>   they risk addressing multiple issues at once, making it harder to
>   address everything comprehensively. Short, concise messages are more
>   efficient and help prevent communication misunderstandings.

Reviewer: What about something like
```suggestion
- **Be considerate of others' resources, including time and attention**
  Long messages, for example, take more time for others to read and respond to, and
  make it harder to determine what the most important points are,
  risking miscommunication when multiple participants have different
  views about what issue is being discussed. Short, concise messages are often more
  efficient and help prevent communication misunderstandings.
```
?

- **Avoid pressing for answers.**
  The Coq community is generally highly responsive to users and
  developers questions. However, in case you don't receive

>   an answer to your interogation, there is no need to push for

Reviewer: ```suggestion
  an answer to your interrogation, there is no need to push for
```

>   response. 

Reviewer: I imagine people reading this will not see their own questions as an interrogation, and this sentence reads to me as assuming by default that the reader is wrong.  I'd rather see a more constructive version ("If you feel your questions or concerns are not receiving timely / adequate responses, instead of taking it up with the individual developers, who may feel XYZ, instead please use ABC process to raise your concerns.  If you experience someone else's questions or messages as excessively JKL, you can use process DEF.")

Reviewer 2: See my response in https://github.com/coq/coq/pull/19812#discussion_r1832381658.

  constructively. Remember that we are different. Different people
  have different perspectives on issues. Being unable to understand why someone
  holds a viewpoint doesn't mean that they're wrong.
- **Be considerate of others' resources, including time and attention**
  Long messages, for example, take more time for others to read and respond to, and
  make it harder to determine what the most important points are,
  risking miscommunication when multiple participants have different
  views about what issue is being discussed. Short, concise messages are often more
  efficient and help prevent communication misunderstandings.
- **Avoid pressing for answers.**
  The Coq community is generally highly responsive to users and
  developers questions. However, in case you don't receive
  an answer to your interrogation, there is no need to push for

>   response. 

Reviewer 2: If you want to mention where to get answers, maybe the Coq Call could be mentioned as a solution (for development-related questions). For users, I don't think there needs to be a specific channel where you are guaranteed an answer.

>   response. 

Reviewer 3: This seems factually incorrect, a question may be missed in which case asking again is the correct action

Reviewer 2: What about clarifying that asking again once is OK, but what is not is to press for answers by repeatedly asking or pinging? Or do you think this is already covered by the point about harassment being when repeated actions that are OK individually become problematic as a pattern.

Reviewer 3: >do you think this is already covered by the point about harassment being when repeated actions that are OK individually become problematic as a pattern.

I guess so.

Reviewer 2: I'm fine with dropping this point, then.

Author: I have dropped this point.

  bad day, so please take this opportunity to question yourself, cool down if
  necessary and do not persist in the exact same behavior you have been
  reported for.

> - **Avoid using personal (member-to-member) communication.** 

Reviewer: I'm concerned phrasing it like this will have a chilling effect on people who have concerns about behaviors.  What do you think of something like
```suggestion
- **Default to using public rather than personal (member-to-member) communication.**
```

  Personal communication channels (e.g., direct messages on Zulip or
  personal emails) limit the visibility of interactions for other users and
  contributors, preventing them from benefiting from shared

>   knowledge. Additionally, unsollicted private communication can

Reviewer: ```suggestion
  knowledge. Additionally, unsolicited private communication can
```

>   sometimes be perceived as too intrusive.

Reviewer: I'd like to see some process in place for community members who are not comfortable being public with initial concerns, e.g., either restricting this to "repeated unsolicited private communication", or saying something like "if you have a concern you're uncertain about sharing publicly, you can reach out to people XYZ"

Reviewer 2: What kind of concerns do you have in mind? Because obviously, to report problematic behaviors, you should not do it in public, but using the coq-conduct mailing list instead. So, probably, you mean something else?

Reviewer: Ah, sorry, I had just missed reading the part of the surrounding text that mentioned the coq-conduct mailing list.

> - **Be mindful when repeated actions become harassment.**

Reviewer: ```suggestion
- **Be mindful to avoid repeated actions that form a pattern of harassment.**
```
